apps/cacvote-server/backend/src/session.rs

```diff
@@ -0,0 +1,156 @@
+use std::fmt::Display;
+
+use auth_rs::{card_details::extract_field_value, certs::VX_CUSTOM_CERT_FIELD_JURISDICTION};
+use axum::{
+    async_trait,
+    extract::FromRequestParts,
+    http::{request::Parts, StatusCode},
+    RequestPartsExt,
+};
+use axum_extra::{
+    headers::{authorization::Bearer, Authorization},
+    TypedHeader,
+};
+use openssl::x509::X509;
+use types_rs::cacvote;
+use uuid::Uuid;
+
+use crate::state::AppState;
+
+/// Represents a simple user session that is kept in memory.
+#[derive(Debug, Clone)]
+pub(crate) struct Session {
+    /// The client's signing certificate signed by the CA that `cacvote-server` trusts.
+    // FIXME: do we need this?
+    #[allow(dead_code)]
+    certificate: X509,
+
+    /// The jurisdiction code of the client's signing certificate.
+    jurisdiction_code: cacvote::JurisdictionCode,
+
+    /// The session token. This is meant to be opaque to the client.
+    token: Uuid,
```

Comment:
I'm using a UUID because it was easy, but I could use something else if we think it'd be better.

Reply:
In terms of length a UUID is fine, what matters is that it's randomly generated 128 bits.